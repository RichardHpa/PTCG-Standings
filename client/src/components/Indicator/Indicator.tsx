export const Indicator = () => {
  return (
    <span className="relative flex h-4 w-4" role="status">
      <span className="absolute inline-flex h-full w-full animate-ping rounded-full bg-green-400 opacity-75"></span>
      <span className="relative inline-flex h-4 w-4 rounded-full bg-green-500"></span>
    </span>
  );
};
