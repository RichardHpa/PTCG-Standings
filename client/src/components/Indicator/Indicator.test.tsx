import { describe, expect, test } from 'vitest';
import { render, screen } from 'tests/testingLibrary';

import { Indicator } from './Indicator';

describe('Indicator', () => {
  test('renders the component', () => {
    render(<Indicator />);
    expect(screen.getByRole('status')).toBeInTheDocument();
  });

  test('matches the snapshot', () => {
    const { asFragment } = render(<Indicator />);
    expect(asFragment()).toMatchSnapshot();
  });
});
