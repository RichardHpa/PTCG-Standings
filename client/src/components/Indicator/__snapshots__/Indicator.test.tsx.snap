// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Indicator > matches the snapshot 1`] = `
<DocumentFragment>
  <span
    class="relative flex h-4 w-4"
    role="status"
  >
    <span
      class="absolute inline-flex h-full w-full animate-ping rounded-full bg-green-400 opacity-75"
    />
    <span
      class="relative inline-flex h-4 w-4 rounded-full bg-green-500"
    />
  </span>
</DocumentFragment>
`;
